codemods/is-builtin-module/index.js

```diff
@@ -17,23 +17,70 @@ export default function (options) {
 			const root = j(file.source);
 			let dirtyFlag = false;
 
-			// Replace import statement
-			root
-				.find(j.ImportDeclaration, {
-					source: { value: 'is-builtin-module' },
-				})
-				.forEach((path) => {
-					path.value.source.value = 'node:module';
-					path.value.specifiers = [
-						j.importSpecifier(j.identifier('isBuiltin')),
-					];
-					dirtyFlag = true;
-				});
+      const importDeclaration = root.find(j.ImportDeclaration, {
+        source: { value: 'is-builtin-module' },
+      })
+
+      const requireDeclaration = root.find(j.VariableDeclarator, {
+        init: {
+          callee: {
+            name: 'require',
+          },
+          arguments: [
+            {
+              value: "is-builtin-module",
+            },
+          ],
+        },
+      })
+
+      // These cases are too complex and too niche to handle
+      if (importDeclaration.size() >= 1 && requireDeclaration.size() >= 1) {
+        console.warn('[WARNING] Refusing to replace `is-builtin-module` because the code mixes import and require statements. Please only use either `import` or `require` and re-run this codemod.')
+        return file.source
+      }
+
+      if (importDeclaration.size() > 1 || requireDeclaration.size() > 1) {
+        console.warn('[WARNING] Refusing to replace `is-builtin-module` because it is imported multiple times. Please import it only once and re-run this codemod.')
+        return file.source
+      }
```

Comment:
These cases are incredibly rare, they would only happen if a dev is doing something wrong. I figured it's not worth to add more complexity to the codemod to handle them and we can safely work with only one import per file.

Reply:
Yeah that sounds fine by me 👍